keras_cv/ops/box_matcher.py

```diff
@@ -0,0 +1,235 @@
+       be assigned to positive_value if its argmax result is above
+       positive_threshold; the rows will be assigned to negative_value if its
+       argmax result is below negative_threshold, and the rows will be assigned
+       to ignore_value if its argmax result is between negative_threshold and
+       positive_threshold.
+    2) thresholds=[negative_threshold, positive_threshold], and
+       match_values=[ignore_value, negative_value, positive_value]: the rows will
+       be assigned to positive_value if its argmax result is above
+       positive_threshold; the rows will be assigned to ignore_value if its
+       argmax result is below negative_threshold, and the rows will be assigned
+       to negative_value if its argmax result is between negative_threshold and
+       positive_threshold.
+    3) thresholds=[positive_threshold], and
+       match_values=[negative_values, positive_value]: the rows will be assigned to
+       positive value if its argmax result is above positive_threshold; the rows
+       will be assigned to negative_value if its argmax result is below
+       negative_threshold.
+
+    Usage:
+
+    ```python
+    box_matcher = keras_cv.ops.ArgmaxBoxMatcher([0.3, 0.7], [-1, 0, 1])
+    iou_metric = keras_cv.bounding_box.compute_iou(anchors, gt_boxes)
+    matched_columns, matched_match_values = box_matcher(iou_metric)
+    cls_mask = tf.less_equal(matched_match_values, 0)
+    ```
+
+    TODO(tanzhenyu): document when to use which mode.
+
+    """
+
+    def __init__(
+        self,
+        thresholds: List[float],
+        match_values: List[int],
+        force_match_for_each_col: bool = False,
+    ):
+        """Constructs ArgmaxBoxMatcher.
+
+        Args:
+          thresholds: A sorted list of floats to classify the matches into
+            different results (e.g. positive or negative or ignored match). The
+            list will be prepended with -Inf and and appended with +Inf.
+          match_values: A list of integers representing matched results (e.g.
+            positive or negative or ignored match). len(`match_values`) must
+            equal to len(`thresholds`) + 1.
+          force_match_for_each_col: each row will be argmax matched to at
+            least one column. This means some columns will be matched to
+            multiple rows while some columns will not be matched to any rows.
+            Filtering by `thresholds` will make less columns match to positive
+            result. Setting this to True guarantees that each column will be
+            matched to positive result to at least one row.
+
+        Raises:
+          ValueError: if `thresholds` not sorted or
+            len(`match_values`) != len(`thresholds`) + 1
+        """
+        if not all([lo <= hi for (lo, hi) in zip(thresholds[:-1], thresholds[1:])]):
+            raise ValueError("`threshold` must be sorted, got {}".format(thresholds))
```

Comment:
let's use f"`threshold` must be sorted, got {thresholds}", same for the other error messages.

Reply:
Done.